### PR TITLE
ASoC: Intel: cht_bsw_rt5672: check return value

### PR DESCRIPTION
**Pull request by aiChaoSONG**
Set codec sysclk could fail and return error, add
error check for it.

### CODE REVIEW

File: sound/soc/intel/boards/cht_bsw_rt5672.c [hunk 1]
```diff
@@ -93,8 +93,12 @@ static int platform_clock_control(struct snd_soc_dapm_widget *w,
 		 * when codec is runtime suspended. Codec needs clock for jack
 		 * detection and button press.
 		 */
-		snd_soc_dai_set_sysclk(codec_dai, RT5670_SCLK_S_RCCLK,
+		ret = snd_soc_dai_set_sysclk(codec_dai, RT5670_SCLK_S_RCCLK,
 				       48000 * 512, SND_SOC_CLOCK_IN);
+		if (ret < 0) {
+			dev_err(card->dev, "failed to set codec sysclk: %d\n", ret);
+			return ret;
+		}
```
Comment (ujfalusi) [inline sound/soc/intel/boards/cht_bsw_rt5672.c:101]: Should we still do the clk_disable_unprepare() even in error case?

Comment (aiChaoSONG) [reply sound/soc/intel/boards/cht_bsw_rt5672.c:101]: I don't think so, there's other calls to `snd_soc_dai_set_sysclk()` in the same file and in other drivers, none of them call `clk_disable_unprepare()` on failure. All just print and return error.

Comment (bardliao) [reply sound/soc/intel/boards/cht_bsw_rt5672.c:101]: @ujfalusi @aiChaoSONG I think `clk_disable_unprepare()` is still needed when `snd_soc_dai_set_sysclk()` return error. 
Turn on/off the clock is the main purpose of `platform_clock_control()`.

Comment (plbossart) [reply sound/soc/intel/boards/cht_bsw_rt5672.c:101]: we have a similar behavior in two other machine drivers:

````c
		ret = snd_soc_dai_set_sysclk(codec_dai, RT5640_SCLK_S_RCCLK,
					     48000 * 512,
					     SND_SOC_CLOCK_IN);
		if (!ret)
			clk_disable_unprepare(priv->mclk);
````
````c
               ret = snd_soc_dai_set_sysclk(codec_dai, RT5645_SCLK_S_RCCLK,
					48000 * 512, SND_SOC_CLOCK_IN);
		if (ret < 0) {
			dev_err(card->dev, "can't set codec sysclk: %d\n", ret);
			return ret;
		}

		clk_disable_unprepare(ctx->mclk);
````

so I vote to align and think later if this makes sense.

**Review by plbossart — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

**Pull request merged**